### PR TITLE
Refactor of Java CGOL example

### PR DESCRIPTION
**Pull request by falkecarlsen**
Refactored object encapsulation of cell away and local optimisations result in 36% load reduction

### CODE REVIEW

File: game_of_life-java/src/main/java/GameOfLife/Canvas.java [hunk 1]
```diff
@@ -6,38 +6,38 @@
...
         // temp canvas-array for swapping before return
-        Cell[][] temp;
+        boolean[][] temp;
         for (int i = 0; i < this.x; i++) {
             for (int j = 0; j < this.y; j++) {
                 neighbours = countMooreNeighbours(i, j);
 
                 // set new state in tempCanvas based on the life of the given cell in canvas
-                tempCanvas[i][j].setState(cellLife(canvas[i][j], neighbours));
+                tempCanvas[i][j]=(cellLife(canvas[i][j], neighbours));
```
Comment (jenrik) [inline game_of_life-java/src/main/java/GameOfLife/Canvas.java]: missing space around assignment operator

File: game_of_life-java/src/main/java/GameOfLife/Graphics.java [hunk 2]
```diff
@@ -24,12 +24,14 @@ public void windowClosing(WindowEvent windowEvent) {
     }
 
     public static void drawCanvas(Canvas canvas, Frame frame) {
+        boolean[][] grid = canvas.getCanvas();
+        java.awt.Graphics graphics = frame.getGraphics();
```
Comment (jenrik) [inline game_of_life-java/src/main/java/GameOfLife/Graphics.java:28]: Why isn't `java.awt.Graphics` imported?

**Review by jenrik — CHANGES_REQUESTED** [review]

File: game_of_life-java/src/main/java/GameOfLife/Canvas.java (hunk 1 above)
Comment (falkecarlsen) [reply game_of_life-java/src/main/java/GameOfLife/Canvas.java]: Fixed in d984780

File: game_of_life-java/src/main/java/GameOfLife/Graphics.java (hunk 2 above)
Comment (falkecarlsen) [reply game_of_life-java/src/main/java/GameOfLife/Graphics.java:28]: It certainly is in line 3, however, the class is called `Graphics, the FQN of the class must be used to differentiate.

**Pull request merged**